Assets/_BForBoss/_Weapons/Scripts/BulletSpawner.cs

```diff
@@ -0,0 +1,52 @@
+using Perigon.Utility;
+using Sirenix.OdinInspector;
+using UnityEngine;
+
+namespace Perigon.Weapons
+{
+
+    public enum BulletTypes
+    {
+        RifleNoPhysics,
+        RiflePhysics
+    }
+    
+    public class BulletSpawner : MonoBehaviour
+    {
+        [SerializeField] private BulletBehaviour[] _bulletPrefabs;
+        [InlineEditor]
+        [SerializeField] private BulletPropertiesScriptableObject _bulletPropertiesScriptableObject;
```

Comment:
Can you set it up so that the Bullet itself have the scriptable object?

Reply:
Actually rethinking about this. 
WeaponBehaviour -> right now includes a bullet spawner. 

Ideally, the weaponScriptableObject would include the bullet it wants to spawn and shoot out.
So i would think the weapon would pass in the desired bullet into the Bullet spawner

Reply:
Sure, so move it out of initialize and then just have it set itself up on awake? Should I just get rid of the interface in that case?

Reply:
Thinking about this further, I'm leaning towards your initial suggestion of having the bullet prefab hold its own scriptable and then having `weaponScriptableObject` hold a bullet type, since that would make for the most robust & self contained object pools.

I could also make it so `weaponScriptableObject` holds the bullet scriptable, and then passing that in every time a new bullet is spawned, but I think we'd start to lose any benefits to that if we decide a specific weapon needs a specific bullet prefab

Reply:
So your idea is to have 
1. Separate enum for bulletType on the weapon
2. Whatevre bulletType is passed into bulletSpawner, the BulletSpawner creates the prefab which contains the bulletScriptable.

The issue is that, whenever I want to create a new bullet to iterate/test/develop, i would need to update the enum. Which can cause issue since you'll need to keep track of the Enum as well as the BulletPrefab + Scriptable

Reply:
Well you wouldn't necessarily run into too many issues there, internally the enum is just an index in the array of prefabs. If you're iterating you just need to make sure you have the same number of enum options and prefabs, and I can make a custom editor to enforce that

Reply:
I'm just trying to avoid coupling bullets to weapons too tightly since it feels redundant to have multiple object pools for identical weapons

Reply:
If you'd prefer, the enum was really only for readability. We could just use an int directly and then the issue of needing to update the enum goes away

Reply:
alright that makes sense. 
Would it be more visually easier to map, if you either have a serialized Dictionary<enum, Prefab> 
or a serializable struct with the enum and prefab.

Reply:
Went ahead and made a custom inspector, this also ensures the array is always the correct length and to the end user doesn't show as an array

Reply:
Okay if this is the case, can you make NoPhysicsBullet its own prefab instead of a variation of the original
![image](https://user-images.githubusercontent.com/5779541/145075129-03e7bbfa-3113-4c03-9543-e98c3a7b511f.png)

Reply:
Just to make sure we're on the same page, Depending on the designs of bullets/guns. we would have 
Shotgun bullets
rocket bullets
Automatic bullets
Single shot rifle bullets
pistol bullets
sniper bullets.

 So the enum would potentially include these in the future.

Reply:
I'd say yes, if those bullets are fundamentally different (use different bullet behaviour scripts or a different prefab). If we can reuse the same bullet for multiple weapons we should wherever possible

Reply:
I'm thinking for mostly the visual appearance of the bullets could be different

Reply:
Yeah in that case they'll be different types

Reply:
@m4rtint just submitted a fix for the prefab, both bullets are their own root prefabs now